src/modules/launcher/PowerLauncher/Helper/NativeEventWaiter.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Windows;
+
+namespace PowerLauncher.Helper
+{
+    public class NativeEventWaiter
+    {
+        public NativeEventWaiter(string eventName, Action callback)
```

Comment:
This method seems like it would be reusable in the future, so it could be moved to ManagedCommon.

Reply:
Interesting idea, but what about `Application.Current.Dispatcher.Invoke(callback);`? This won't work if it's in ManagedCommon.

Reply:
I suppose we could wrap `Application.Current.Dispatcher.Invoke` around the `onHotkey` method while declaring the _hotkeyEventWaiter